### ISSUE TITLE
Classes defined in the __main__ module break Object construction process in v0.2

### ISSUE THREAD

**Issue by krafczyk** [ncsa/DRYML#21]
When trying to save an object whose class is defined in `__main__`, and then loading from disk, the incorrect object signature is loaded and the load fails. Example code triggering the issue:

```
from dryml import Object, load_object

# We create a new very simple class just to remember data
# in its arguments
class Data(Object):
    def __init__(self, data):
        pass

# Construct a new instance of Data
obj = Data([1, 2, 3, 4, 5])

obj.save_self('obj.dry')

new_obj = load_object('obj.dry') # Fails
```

The failure looks like this:
```
Traceback (most recent call last):
  File "/home/matthew/Projects/DRYML/dryml/test_bug.py", line 18, in <module>
    new_obj = load_object('obj.dry')
              ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/matthew/Projects/DRYML/dryml/src/dryml/object.py", line 416, in load_object
    raise RuntimeError(
RuntimeError: Loaded object doesn't have expected definition!
expected: {'cls': <class '__main__.Data'>, 'dry_mut': False, 'dry_args': ([1, 2, 3, 4, 5],), 'dry_kwargs': {'dry_id': '9d44583e-cd40-41ff-8e5f-c60a12b33cc8', 'dry_metadata': {'description': '', 'creation_time': 1724525061.1574423}}}
got: {'cls': <class '__main__.Data'>, 'dry_mut': False, 'dry_args': ([1, 2, 3, 4, 5],), 'dry_kwargs': {'dry_id': 'd571dea1-f703-4ce2-8e11-2ba53a7f9cc0', 'dry_metadata': {'description': '', 'creation_time': 1724525061.1721513}}}
```

So we can see multiple problems here. The `dry_id` didn't get transferred, and neither did the creation time.

This issue dissapears when we move the class definition to a separate file/module and load it that way. Notably, none of the tests are currently picking up this particular failure.
```

**Comment by krafczyk** [comment]
This issue seems to affect all versions of dryml I've tested.

**Comment by krafczyk** [comment]
Fixed by #22
